src/main/java/puke/TaskList.java

```diff
@@ -0,0 +1,159 @@
+package puke;
+
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+import puke.exceptions.TaskNumberOutOfBoundsException;
+import puke.task.Deadline;
+import puke.task.Event;
+import puke.task.Task;
+import puke.task.Todo;
+
+/**
+ * Manages a list of tasks, providing methods to add, delete, and manipulate tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Constructs a TaskList with tasks loaded from storage.
+     *
+     * @param loadedFromStorage the tasks loaded from storage
+     */
+    public TaskList(ArrayList<Task> loadedFromStorage) {
```

Comment:
Nice use of descriptive variable names; it really improves the readability of the code.